Dockerfile

```diff
@@ -0,0 +1,47 @@
+#### Base image ####
+
+# A base image from which other images can be built.
+FROM python:3.8.6-buster as base
+
+# Install Poetry.
+RUN pip3 install poetry==1.1.2
+
+# Create dedicated application folder in image and move into it.
+RUN mkdir /app/
+WORKDIR /app/
+
+# Copy dependency congiguration files from host system into a dedicated application folder.
+COPY \
+    # Dependency configuration files.
+    poetry.lock \
+    poetry.toml \
+    pyproject.toml \
+    # Copy them to the application folder.
+    ./
+
+# Install application dependencies.
+RUN poetry install
+
+# Copy source code files from host system into a dedicated application folder.
+COPY ./app/ ./app/
```

Comment:
To avoid confusion it might be worth renaming the `app` folder to `todo_app`

Reply:
Do we need to copy this over for the development container (considering we're going to bind mount it anyway?)